docs/topics/multiplatform/multiplatform-advanced-project-structure.md

```diff
@@ -168,6 +168,36 @@ There are three important concepts in dependency resolution:
 
    ![Error on JVM-specific API in common code](dependency-resolution-error.png){width=700}
 
+### Aligning versions of common dependencies across source sets
+
+In Kotlin Multiplatform projects, the common source set is compiled several times: to produce a klib, and as a part of each
+configured [compilation](multiplatform-configure-compilations.md). To produces consistent binaries, common code
+should be compiled against the same versions of multiplatform dependencies each time.
+Kotlin Gradle plugin helps to align these dependencies, making sure that the effective dependency version
+is the same for each source set.
+
+In the example above, imagine that you want to add the `androidx.navigation:navigation-compose:2.7.7` dependency to your
+`androidMain` source set. Your project explicitly declares the `kotlinx-coroutines-core:1.7.3` dependency for the `commonMain`
+source set, but the Compose Navigation library with the version 2.7.7 requires Kotlin coroutines 1.8.0 or newer.
+
+Since `commonMain` and `androidMain` are compiled together, Kotlin Gradle plugin chooses between the two version of the
+coroutines library and applies `kotlinx-coroutines-core:1.8.0` to the `commonMain` source set. But to make the common code
+compile consistently across all configured targets, the iOS source sets also need to be constrained to the same dependency version.
+So Gradle propagates the `kotlinx.coroutines-*:1.8.0` dependency to the `iosMain` source set as well.
+
+![Alignment of dependencies in a group of source sets](multiplatform-source-set-dependency-alignment.svg){width=700}
+
+Dependencies are aligned across source sets of each group: the `*Main` source sets and the [`*Test` source sets](multiplatform-discover-project.md#integration-with-tests).
```

Comment:
We haven't previously used the term 'group of source sets', and I found it a bit confusing, because 'the group of test source sets' sounds like a part of all test source sets

I'd recommend to write it explicitly each time:

> The Gradle configuration for *Test source sets includes all dependencies of *Main source sets but not vice versa.

> For example, you have the Kotlin coroutines 1.7.3 dependency in your *Main source sets, propagated to every project source set.